opm/common/utility/numeric/VectorUtil.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef VECTORUTIL_H
+#define VECTORUTIL_H
+
+#include <tuple>
+#include <vector>
+#include <functional>
+#include <stdexcept> 
+#include <type_traits>
+
+namespace VectorUtil {
+
+// Implementation of generation General Operation between two vectors of the same type
+template <typename T, typename Operation>
+std::vector<T> vectorOperation(const std::vector<T>& vecA, const std::vector<T>& vecB, Operation op) {
+    if (vecA.size() != vecB.size()) {
+        throw std::invalid_argument("Error: Vectors must have the same size!"); // Throwing exception
+    }
+    std::vector<T> result;
+    result.reserve(vecA.size());
+    // Use std::transform with the passed operation
+    std::transform(vecA.begin(), vecA.end(), vecB.begin(), std::back_inserter(result), op);
+    return result;
+}
+
+
+// A simple utility function to calculate area of a rectangle
+std::tuple<std::array<double,4>, std::array<double,4>, std::array<double,4>> 
+appendNode(const std::array<double,3>&, const std::array<double,3>&, 
+           const std::array<double,3>&, const double&, const double&, 
+           const double&);
+template <typename T>
+std::vector<T> filterArray(const std::vector<std::size_t>& X, const std::vector<int>& ind){
+    std::vector<T> filtered_vectorX(ind.size(),0);
+    for (std::size_t index = 0; index < ind.size(); index++) {
+        filtered_vectorX[index] = X[ind[index]];
+    }
+    return filtered_vectorX;
+};       
+
+// T type of the object
+// Rout type of the object output 
+// Method type of method
+template <typename T,  typename Rout, typename Rin = std::size_t, typename Method, typename... Args>
+std::vector<Rout> callMethodForEachInputOnObject(const T& obj, Method mtd, const std::vector<Rin>& input_vector, Args&&... args) {
+    std::vector<Rout> result;
+    // Reserve space for each vector in the tuple
+    result.reserve(input_vector.size());
+    // Iterate over the input_vector and fill the tuple's vectors
+    for (const auto& element : input_vector) {
+        Rout value = (obj.*mtd)(element, std::forward<Args>(args)...);
```

Comment:
Please note that when calling `std::forward<>()` in a loop you risk moving from (some of the) `args` twice.  That's undefined behaviour.   I think we'll have to be conservative here and just use
```C++
Rout value = (obj.*mtd)(element, args...);
```
instead.